### ISSUE TITLE
[lib.dom.d.ts] `hashchange` in `WindowEventHandlersEventMap` maybe wrong?

### ISSUE THREAD

**Issue by ice-zjchen** [microsoft/TypeScript#45872]
# lib Update Request

## Configuration Check

My compilation *target* is `ES2015` and my *lib* is `the default`.

## Missing / Incorrect Definition

```typescript

interface WindowEventHandlersEventMap {
    "afterprint": Event;
    "beforeprint": Event;
    "beforeunload": BeforeUnloadEvent;
    "gamepadconnected": Event;
    "gamepaddisconnected": Event;

    // HERE
    "hashchange": Event;

    "languagechange": Event;
    "message": MessageEvent;
    "messageerror": MessageEvent;
    "offline": Event;
    "online": Event;
    "pagehide": PageTransitionEvent;
    "pageshow": PageTransitionEvent;
    "popstate": PopStateEvent;
    "rejectionhandled": PromiseRejectionEvent;
    "storage": StorageEvent;
    "unhandledrejection": PromiseRejectionEvent;
    "unload": Event;
}

```

## Sample Code

```typescript
export function addUrlChangeListen(
  hashChangeListener: (ev: HashChangeEvent) => void,
  stateChangeListener?: (ev?: PopStateEvent) => void,
): () => void {

  // the type here is HashChangeEvent in previous version. but now it is Event
  window.addEventListener('hashchange', hashChangeListener);
}
```

`HashChangeEvent` !== `Event`. and in my opinion, it should be HashChangeEvent

## Documentation Link

relate PR: https://github.com/microsoft/TypeScript/blame/main/lib/lib.dom.d.ts#L17315

**Comment by MartinJohns** [comment]
See also: https://github.com/microsoft/TypeScript-DOM-lib-generator/

**Comment by sandersn** [comment]
This is fixed in 4.5.
